src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -37,6 +37,22 @@ public static bool IsReservedKeyword(SyntaxKind kind)
             return kind >= SyntaxKind.BoolKeyword && kind <= SyntaxKind.ImplicitKeyword;
         }
 
+        public static bool IsControlKeyword(SyntaxKind kind)
+        {
+            return kind == SyntaxKind.YieldKeyword
+                || kind == SyntaxKind.WhenKeyword
+                || kind == SyntaxKind.InKeyword
+                || (kind != SyntaxKind.LockKeyword 
+                    && kind >= SyntaxKind.IfKeyword 
+                    && kind <= SyntaxKind.ThrowKeyword);
+        }
+
+        public static bool IsControlStatement(SyntaxKind kind)
+        {
+            return (kind >= SyntaxKind.GotoStatement && kind <= SyntaxKind.ForEachStatement) // Jump Statements
+                || (kind >= SyntaxKind.IfStatement && kind <= SyntaxKind.FinallyClause); // Checked Statements
```

Comment:
1. same.   i have no idea what these ranges are, or if the values are appropriate.  can we just enumerate them so we can discuss if they make sense as 'control' statements?
2. does the language have a concept of 'ControlStatmenet' if not, i'm not sure how i feel about this being at the compilation layer.